lib/queue.js

```diff
@@ -621,6 +626,9 @@ Queue.prototype.add = function(name, data, opts){
     data = name;
     name = Job.DEFAULT_JOB_NAME;
   }
+  if (!opts) opts = {};
+  _.defaults(opts, this.defaultJobOptions);
```

Comment:
why merge the job options into the standard opts object? besides it may create side effects since it is modifying inputs.

Reply:
This is in `Queue.prototype.add`, this is not the main queue options. `opts` here refers to the options passed to `Queue.add(data, opts)`, not `new Queue(opts)`, so I'm merging the default job options into the passed job options.